ipatests/test_integration/test_trust.py

```diff
@@ -993,3 +1036,248 @@ def test_server_option_with_unreachable_ad(self):
+                "idoverrideuser-mod",
+                "Default Trust View",
+                "--uid=88888888",
+                "--gid=88888888",
+                nonposixuser,
+            ]
+        )
+        tasks.clear_sssd_cache(self.master)
+        self.mod_idrange_auto_private_group("true")
+        tasks.clear_sssd_cache(self.clients[0])
+        result = self.clients[0].run_command(['id', nonposixuser])
+        uid_reg = (
+            r"^uid=(88888888)\({0}\)".format
+            (nonposixuser)
+        )
+        gid_reg = r"gid=(88888888)"
+        grp_reg = (
+            r"groups=(88888888),(\d+)\({0}@{1}\)".format
+            (domainusers, self.ad_domain)
+        )
+        assert re.search(uid_reg, result.stdout_text)
+        assert re.search(gid_reg, result.stdout_text)
+        assert re.search(grp_reg, result.stdout_text)
+
+    @pytest.mark.xfail(reason='bz2010512', strict=True)
+    def test_idoverride_with_auto_private_group_hybrid(self):
+        """
+        override ad trusted user in default trust view
+        and set auto-private-groups=hybrid and ensure that
+        overridden values takes effect.
+        """
+        domainusers = 'domain users'
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        self.master.run_command(
+            [
+                "ipa",
+                "idoverrideuser-mod",
+                "Default Trust View",
+                "--uid=78888888",
+                "--gid=79999999",
+                nonposixuser
+            ]
+        )
+        tasks.clear_sssd_cache(self.master)
+        self.mod_idrange_auto_private_group("hybrid")
+        tasks.clear_sssd_cache(self.master)
+        tasks.clear_sssd_cache(self.clients[0])
+        result = self.clients[0].run_command(['id', nonposixuser])
+        uid_reg = (
+            r"^uid=(78888888)\({0}\)".format
+            (nonposixuser)
+        )
+        gid_reg = r"gid=(79999999)"
+        grp_reg = (
+            r"groups=(78888888),(\d+)\({0}@{1}\)".format
+            (domainusers, self.ad_domain)
+        )
+        assert re.search(uid_reg, result.stdout_text)
+        assert re.search(gid_reg, result.stdout_text)
+        assert re.search(grp_reg, result.stdout_text)
+
+    def test_nonposixuser_nondefault_primary_group(self):
+        """
+        Test for non default primary group
+        """
+        nonposixuser = "nonposixuser1@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group("true")
+        tasks.clear_sssd_cache(self.master)
+        self.clients[0].run_command(['id', nonposixuser])
+        # Remove trust
+        self.remove_trust(self.ad)
+
+
+class TestPosixAutoPrivateGroup(BaseTestTrust):
+    """
+    Tests for auto-private-groups option with non posix AD trust
+    """
+    topology = 'line'
+    num_ad_domains = 1
+    num_clients = 1
+
+    def test_add_posix_trust(self):
+        self.add_trust('ipa-ad-trust-posix')
+
+    @pytest.mark.xfail(reason='bz2010512', strict=True)
```

Comment:
The default setting for posix trust is "false", hence this test should not be able to resolve the user. Don't mark it as xfail but rather expect an exception "no such user"